### PR TITLE
Infra/live code playground device wrapper

### PR DESCRIPTION
**Pull request by adids1221**
## Description
Docs playground (`live-code`) device wrapper to simulate `Mobile or Tablet` device.
In the `buildDocs` script wrapping the component snippet with the `MobileDeviceWrapper` that changes the size of the `View` container according to the selected mode.

Note: in `buildDocs` script there is a new function `processSnippet` that process the component `snippet`, since now we have cases that the live-code playground render a function or plain JSX that should be wrapped with the `MobileDeviceWrapper`.

Based on #3272 

## Changelog
Docs playground (`live-code`) device wrapper to simulate `Mobile or Tablet` device.

## Additional info
MADS-4507

### CODE REVIEW

**Comment by adids1221** [comment]
@M-i-k-e-l I mentioned in the description that `Assets and Incubator` export throw an error in this branch.
If it's working on your PR I should fix that here.

**Comment by M-i-k-e-l** [comment]
@adids1221 please merge the conflicts

**Comment by adids1221** [comment]
> @adids1221 please merge the conflicts

Iv'e merged the conflicts, Assets and Incubator throw an error.

File: docuilib/src/theme/ReactLiveScope/Playground.tsx [hunk 1]
```diff
@@ -24,3 +41,81 @@ export function ToggleControl(props: ToggleControlProps) {
     </View>
   );
 }
+
+export function MobileDeviceWrapper({children}) {
```
[the remark this answers is not in the document]
Comment (adids1221) [reply docuilib/src/theme/ReactLiveScope/Playground.tsx:45]: You will see it in the next commit

File: docuilib/src/theme/ReactLiveScope/index.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ const ReactLiveScope = {
   ...React,
   /* Docs' utils and components */
   Data,
-  Playground,
+  ...Playground,
```
[the remark this answers is not in the document]
Comment (adids1221) [reply docuilib/src/theme/ReactLiveScope/index.js:58]: Revert will be in the next commit.

File: docuilib/src/theme/ReactLiveScope/Playground.tsx [hunk 3]
```diff
@@ -24,3 +41,81 @@ export function ToggleControl(props: ToggleControlProps) {
...
+      <View gap-s1 marginV-s1 center>
+        <Text marginR-s3 style={styles.typesTitle}>
+          Choose Type
+        </Text>
+        <SegmentedControl
+          initialIndex={0}
+          segments={simulatorOptions}
+          onChangeIndex={index => {
+            onPress(simulatorOptions[index].value);
+          }}
```
[the remark this answers is not in the document]
Comment (adids1221) [reply docuilib/src/theme/ReactLiveScope/Playground.tsx:97]: Moved the `onPress` logic into `onChangeIndex`

File: docuilib/src/theme/ReactLiveScope/Playground.tsx [hunk 4]
```diff
@@ -24,3 +41,81 @@ export function ToggleControl(props: ToggleControlProps) {
...
+  };
+
+  return (
+    <View gap-s3>
+      <View gap-s1 marginV-s1 center>
+        <Text marginR-s3 style={styles.typesTitle}>
+          Choose Type
+        </Text>
+        <SegmentedControl
+          initialIndex={0}
```
[the remark this answers is not in the document]
Comment (adids1221) [reply docuilib/src/theme/ReactLiveScope/Playground.tsx]: `findSelectedIndex` uses the current selected type to determine the initial index.

File: scripts/buildDocs.js [hunk 5]
```diff
@@ -38,6 +38,28 @@ fs.mkdirSync(COMPONENTS_DOCS_DIR);
...
+    //Check if <div> tag is immediately after return statement
+    const divTagIndex = processedSnippet.findIndex(line =>
+      line.trim().startsWith('return (') && processedSnippet[processedSnippet.indexOf(line) + 1]?.includes('<div>'));
+
+    if (divTagIndex !== -1) {
+      //If <div> tag is found, replace <div> and </div> with <MobileDeviceWrapper> and </MobileDeviceWrapper>
+      return processedSnippet
+        .map(line => line.replace('<div>', '<MobileDeviceWrapper>').replace('</div>', '</MobileDeviceWrapper>'))
+        .join('\n');
+    }
```
[the remark this answers is not in the document]
Comment (adids1221) [reply scripts/buildDocs.js:57]: Let's sync about it.
Iv'e tried to implement it as HOC but I had some issues.

File: docuilib/yarn.lock [hunk 6]
```diff
@@ -6069,7 +6069,7 @@ lodash.uniq@^4.5.0:
   resolved "https://registry.npmjs.org/lodash.uniq/-/lodash.uniq-4.5.0.tgz#d0225373aeb652adc1bc82e4945339a842754773"
   integrity sha512-xfBaXQd9ryd9dlSDvnvI0lvxfLJlYAZzXomUYzLKtUeOQvOP5piqAWuGtrhWeqaXK9hhoM/iyJc5AV+XfsX3HQ==
 
-lodash@^4.17.20, lodash@^4.17.21:
+lodash@4.x.x, lodash@^4.17.20, lodash@^4.17.21:
```
Comment (M-i-k-e-l) [inline docuilib/yarn.lock:6072]: `yarn.lock` should not change

**Review by M-i-k-e-l — COMMENTED** [review]
This change breaks `SortableGridList`

File: docuilib/src/theme/ReactLiveScope/MobileDeviceWrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  MOBILE = 'MOBILE',
+  TABLET = 'TABLET'
+}
+
+const SimulatorSizes = {
+  [SimulatorTypes.MOBILE]: {width: 340, height: 754},
+  [SimulatorTypes.TABLET]: {width: 900, height: 754}
+};
+
+const simulatorOptions = [
```
[the remark this answers is not in the document]
Comment (adids1221) [reply docuilib/src/theme/ReactLiveScope/MobileDeviceWrapper.tsx:21]: Yes, changed.

File: docuilib/yarn.lock (hunk 6 above)
Comment (adids1221) [reply docuilib/yarn.lock:6072]: You are right I'm not sure about that checking it.

Comment (adids1221) [reply docuilib/yarn.lock:6072]: I didn't managed to revert it, tried to checkout the commit and revert the file didn't managed to.

**Comment by adids1221** [comment]
@M-i-k-e-l Alexey made some changes in the Design.
Also I want to check the option of rendering an Iframe so we can use Picker and other components that extends Modal as in the mobile.

**Comment by adids1221** [comment]
Closing this PR, #3351 has new approach with iframe.

**Pull request closed without merging**